### PR TITLE
8260339: JVM crashes when executing PhaseIdealLoop::match_fill_loop

### PR DESCRIPTION
**Pull request by Wanghuang-Huawei**
The reason is :

```
  BasicType t = store->as_Mem()->memory_type();
  const char* fill_name;
  if (msg == NULL &&
      StubRoutines::select_fill_function(t, false, fill_name) == NULL) {
    msg = "unsupported store";
    msg_node = store;
  }
```

If the `store` is a `StoreVectorNode` ,the `BasicType` is `T_VOID`. It seems that we don't need to intrinsify a `StoreVectorNode` filling here.

I add a new case here to avoid mistake:
```
  case T_NARROWOOP:
  case T_NARROWKLASS:
  case T_ADDRESS:
+  case T_VOID:
    // Currently unsupported
    return NULL;
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260339](https://bugs.openjdk.java.net/browse/JDK-8260339): JVM crashes when executing PhaseIdealLoop::match_fill_loop 

### Reviewers
 * [Nils Eliasson](https://openjdk.java.net/census#neliasso) (@neliasso - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)
 * [Igor Ignatyev](https://openjdk.java.net/census#iignatyev) (@iignatev - **Reviewer**)

### Contributors
 * He Xuejin `<hexuejin2@huawei.com>`

### Download
`$ git fetch https://git.openjdk.java.net/jdk16 pull/132/head:pull/132`
`$ git checkout pull/132`

### CODE REVIEW

**Review by neliasso — COMMENTED** [review]
The fix looks good.

Since JDK 16 is in rampdown phase 2 (http://openjdk.java.net/jeps/3) - this fix will need additional approval before it can be pushed. I will help out with that.

**Comment by Wanghuang-Huawei** [comment]
> The fix looks good.
> 
> Since JDK 16 is in rampdown phase 2 (http://openjdk.java.net/jeps/3) - this fix will need additional approval before it can be pushed. I will help out with that.

Thank you for your review and help.

**Comment by DamonFool** [comment]
Hi @Wanghuang-Huawei , I'm still not clear how to reproduce the bug.

Is it possible to add a jtreg test for this fix?
Thanks.

**Review by vnkozlov — CHANGES_REQUESTED** [review]
Need regression test since change is in common code.

**Comment by iignatev** [comment]
Hi @Wanghuang-Huawei,

given JDK16 is RDP2 and your patch changes the shared code, you are to integrate an automated jtreg test together with the fix. I've taken `TestLoopStoreVector.java` from 8260339 and added the minimal required jtreg test description, and double-checked that the test fails on linux-aarch64 w/o the fix and passes w/ it. you can take the jtreg-ified test from my [pull/132](https://github.com/iignatev/jdk16/tree/pull/132) branch (you will need to replace `$copyright-header` w/ the appropriate line), feel free to use it as-is or modify.

Thanks,
-- Igor

**Comment by DamonFool** [comment]
> Hi @Wanghuang-Huawei , I'm still not clear how to reproduce the bug.
> 

Ah, I missed the attachment in the JBS.
Thumbs-up @Wanghuang-Huawei .

**Comment by Wanghuang-Huawei** [comment]
> Hi @Wanghuang-Huawei,
> 
> given JDK16 is RDP2 and your patch changes the shared code, you are to integrate an automated jtreg test together with the fix. I've taken `TestLoopStoreVector.java` from 8260339 and added the minimal required jtreg test description, and double-checked that the test fails on linux-aarch64 w/o the fix and passes w/ it. you can take the jtreg-ified test from my [pull/132](https://github.com/iignatev/jdk16/tree/pull/132) branch (you will need to replace `$copyright-header` w/ the appropriate line), feel free to use it as-is or modify.
> 
> Thanks,
> -- Igor

Thank you very much. I will push a new commit which contains jtreg test.

**Comment by neliasso** [comment]
@Wanghuang-Huawei Have you been able to reproduce this on any other platform that linux-aarch64? I'm having trouble reproducing this on x64.

**Comment by DamonFool** [comment]
> @Wanghuang-Huawei Have you been able to reproduce this on any other platform that linux-aarch64? I'm having trouble reproducing this on x64.

I tried the reproducer on the JBS just now and couldn't reproduce it on x64 either.

**Comment by nsjian** [comment]
> @Wanghuang-Huawei Have you been able to reproduce this on any other platform that linux-aarch64? I'm having trouble reproducing this on x64.

I think x64 has disabled OptimizeFill by default. I can reproduce this with -XX:+OptimizeFill.

**Comment by neliasso** [comment]
Please add  -XX:+OptimizeFill to the regression test

**Comment by vnkozlov** [comment]
Yes, OptimizeFill  is off after 8247307 changes in JDK 16. Add it to @run command (it is C2 flag so you need `Ignore*` flag):
```
* @run main/othervm -XX:+IgnoreUnrecognizedVMOptions -XX:+OptimizeFill
```

File: test/hotspot/jtreg/compiler/vectorapi/TestLoopStoreVector.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import jdk.incubator.vector.IntVector;
+import jdk.incubator.vector.LongVector;
+import jdk.incubator.vector.VectorSpecies;
+
+/*
+ * @test
+ * @bug 8260339
+ * @modules jdk.incubator.vector
+ *
+ * @run main/othervm -XX:+IgnoreUnrecognizedVMOptions -XX:+OptimizeFill
```
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/vectorapi/TestLoopStoreVector.java]: You forgot test name

**Review by vnkozlov — APPROVED** [review]
Good.
Wait an other day for other people to look.

**Review by iignatev — APPROVED** [review]
LGTM

**Review by neliasso — APPROVED** [review]
Looks good.

**Comment by neliasso** [comment]
I will sponsor this change when 24 hours have passed since the last update.

**Pull request closed without merging**